waiter/src/waiter/token.clj

```diff
@@ -516,48 +516,61 @@
                            :log-level :warn}))))
 
       nil
-      (do
+      (let [existing-editor (get existing-token-metadata "editor")
+            existing-owner (get existing-token-metadata "owner")
+            owner? (and existing-owner (authz/manage-token? entitlement-manager authenticated-user token existing-token-metadata))
+            editor? (and existing-editor (not owner?)  (authz/run-as? entitlement-manager authenticated-user existing-editor))
+            delegated-user (if editor?
```

Comment:
A weird situation can happen when setting an editor on a token that doesn't have an owner.  e.g. If run-as-user is set to me, and owner is null, and I set editor to me as well, I can no longer change the owner, editor, or run-as-user.

Reply:
Do we always have a `run-as-user`?

Reply:
Responded here: https://github.com/twosigma/waiter/pull/1313#discussion_r603327556